src/libstd/panic.rs

```diff
@@ -330,7 +330,7 @@ impl<'a, F: Future> Future for AssertUnwindSafe<F> {
     fn poll(mut self: PinMut<Self>, cx: &mut task::Context) -> Poll<Self::Output> {
         unsafe {
             let pinned_field = PinMut::new_unchecked(
-                &mut PinMut::get_mut(self.reborrow()).0
+                &mut PinMut::get_mut_unchecked(self.reborrow()).0
```

Comment:
Couldn't this use `PinMut::map_unchecked`?

Reply:
Jep. Good point. I changed it